### ISSUE TITLE
DuckDB-CLI crashes with Snap amazon0601 dataset

### ISSUE THREAD

**Issue by SunnyYasser** [duckdb/duckdb#14824]
### What happens?

When executing a complex query with duckdb cli, the process gets terminated and a ```killed``` message is printed.

### To Reproduce

```
 CREATE TABLE RBIG AS SELECT * FROM read_csv_auto('../scratch/Amazon0601.txt');

 ALTER TABLE RBIG RENAME column 'ToNodeId' to dest;

 ALTER TABLE RBIG RENAME column '# FromNodeId' to src;

 SELECT R1.src AS a, R1.dest AS b, R2.dest AS c, R3.dest AS d, R4.dest AS e
  FROM RBIG AS R1
  JOIN RBIG AS R2 ON R1.src = R2.src
  JOIN RBIG AS R3 ON R1.dest = R3.src
  JOIN RBIG AS R4 ON R2.dest = R4.src;

```

### OS:

Ubuntu 24.04.1 LTS  x86_64

### DuckDB Version:

v1.1.2

### DuckDB Client:

CLI

### Full Name:

Sunny Yasser

### Affiliation:

Ecole Polytechnique de Montreal

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hi @SunnyYasser, thanks for filing this issue. This query implements a graph pattern with self joins. A few years ago I wrote a blog post about why these are expensive operations (https://szarnyasg.github.io/posts/self-joins/).

When I run this query on my MacBook Pro with the latest macOS, this query uses approximately 22 GB memory. So it's likely that your instance runs out of memory.

You can check the error code by running:

```bash
echo $?
```

If it returns 137, that means your process ran out of memory.

By default, DuckDB tries to spill to disk both in in-memory mode and in persistent mode but it's worth double-checking the settings. See the guide here: https://duckdb.org/docs/guides/performance/how_to_tune_workloads.html#spilling-to-disk

**Comment by SunnyYasser** [comment]
@szarnyasg thanks for the reply, I checked and the error code was indeed 137. However, I am not sure why disk spilling is not being executed by default. The link you shared explains it happens by default.

**Comment by szarnyasg** [comment]
@SunnyYasser Yes, indeed it should spill to disk. I wonder: if you set it manually, does the query complete successfully?

**Comment by SunnyYasser** [comment]
> @SunnyYasser Yes, indeed DuckDB should spill to disk by default. I wonder: if you set it manually, does the query complete successfully?

No it does not. I tried ```SET temp_directory = 'overflow/';``` before executing but still the same behaviour is observed. I wonder if its CLI specific or not

**Comment by szarnyasg** [comment]
It may be allocator-specific too. Can you try the nightly build?
https://duckdb.org/docs/installation/?version=main&environment=cli&platform=linux&download_method=direct&architecture=x86_64

**Comment by szarnyasg** [comment]
@SunnyYasser how much memory does the machine have?

**Comment by SunnyYasser** [comment]
@szarnyasg ```cat /proc/meminfo | grep MemTotal``` gives:
  ```MemTotal:       32448100 kB```

**Comment by szarnyasg** [comment]
@SunnyYasser, thanks for the information. I could reproduce the issue now.

**Comment by lnkuiper** [comment]
Closed via https://github.com/duckdb/duckdb/pull/19437
